File: OOP/Proyects/De C++ a Java/Pila/C++/Stack.cpp
#include "Stack.h"//incluimos nuestra clase Stack
#include <iostream> //incluimos la libreria iostream

Stack::Stack()//se crea el constructor
{
	TOP=0;//inicializamos top en cero
}

bool Stack::isEmpty(){// creamos una funcion tipo bool isEmpty
	return(TOP==0);//si TOP es igual a cero ,es verdadero,si es diferente de cero es Falso
}

bool Stack::Push(int elemento){//creamos la funcion push de tipo bool donde podremos ingresar un elemento 
	if(isFull())//si la pila ya esta llena
		return false;//retorna falso
	arrStack[TOP]=elemento;//el valor del arrreglo Top es igual al elemento
	TOP++;//top incrementa su valor
}

TipoDeDato Stack::Pop(){//creamos una funcion de tipo de dato pop
	int x;//declaramos una variable x de tipo entero
	
	if(isEmpty())//si la pila esta vacia
		return -1000000;//regresa -1000000
	
	TOP--; //top decrementa su valor
	x=arrStack[TOP];//la variable x es igual al valor de TOP del arreglo
	return x;//regresa el valor de x
}

TipoDeDato Stack::Top(){//creamos una funcion de tipo de dato top
	int x;//declaramos una variable x  de tipo entero
	
	if(isEmpty())//si la pila esta vacia
		return -1000000;//regresa -1000000
	
	x=arrStack[TOP-1];//la variable x es igual al valor en la posicion TOP-1 del arreglo
	return x;//regresa el valor de x
}

bool Stack::isFull(){//creamos la funcion de tipo bool isFull
	return(TOP==TAM);//regresa top y su tamano 
}

File: OOP/Proyects/De C++ a Java/Pila/Java/src/Stack.java

package pilas;//En el package pilas estaran las clases relacionadas entre si

 
public class Stack {
    int TAM = 50;//Declaramos la variablo TAM (tamaño) de tipo entero con un valor de 50
    private int TOP;//Declaramos la variable privada TOP de tipo entero
    private int[] arrStack = new int[TAM];//Declaramos el arreglo privado de tipo entero y el velor del arreglo es igual a TAM
    
    
    public Stack(){//Creamos el constructor
        TOP = 0;//TOP se incializa en 0
    }
    
    public boolean isEmpty(){//Se crea la funcion de tipo boleana isEmpty
        return(TOP==0);//si TOP=0 entonces es verdadero, si no es Falso
    }
    
    public boolean isFull() {//Se crea la funcion de tipo boleana isFull
        return(TOP==TAM);//Si TOP=Tam entonces es verdadero, si no es Falso
    }
    public void push(int e){//Se crea la funcion push de tipo void, en la cual se ingresara un valor de tipo entero
        if(isFull()){//Si la pila esta llena
            System.out.print("Pila llena");//Mostrar pila llena 
        }//Si no
        arrStack[TOP]=e;//ingresa e en el arreglo en la posicion TOP
        TOP++;//Top incrementa su valor 
    }

    public int pop(){//Se crea la funcion pop de tipo entero
        int x;//Se declara la variable x de tipo enter
        
        if(isEmpty()){//Si la pila esta vacia
            return -10000000;//Regresa -1000000
        }//si no
        TOP--;//TOP decrementa su valor
        x = arrStack[TOP];//x es igual a el valor del arreglo que este en la posicio TOP
        return x;//Regresa el valor de x
    } 
   
    public int TOP(){//Se crea la funcion Top de tipo entero
        int x;//Se declara la variable x de tipo entero
        
        if(isEmpty()){//Si la pila esta vacia
            System.out.print("Pila vacia");//Mostramos que la pila esta vacia
        }//si no
        
        x = arrStack[TOP-1];//x es igual al valor en la posicion TOP-1 del arreglo
        return x;//Regresa el valor de x
    }
 
    
    
}
